tests/python/pytorch/graphbolt/impl/test_ondisk_dataset.py

```diff
@@ -1711,6 +1711,42 @@ def test_OnDiskDataset_load_tasks():
         dataset = None
 
 
+def test_OnDiskDataset_load_all_nodes_set():
+    """Test preprocess of OnDiskDataset."""
+    with tempfile.TemporaryDirectory() as test_dir:
+        # All metadata fields are specified.
+        dataset_name = "graphbolt_test"
+        num_nodes = 4000
+        num_edges = 20000
+        num_classes = 10
+
+        # Generate random graph.
+        yaml_content = gbt.random_homo_graphbolt_graph(
+            test_dir,
+            dataset_name,
+            num_nodes,
+            num_edges,
+            num_classes,
+        )
+        yaml_file = os.path.join(test_dir, "metadata.yaml")
+        with open(yaml_file, "w") as f:
+            f.write(yaml_content)
+
+        dataset = gb.OnDiskDataset(test_dir).load()
+
+        all_nodes_set = dataset.all_nodes_set
+
+        # assert (False), f"{dataset.graph.num_nodes}, {(all_nodes_set)}"
+
+        if isinstance(all_nodes_set, gb.ItemSet):
+            for i, item in enumerate(all_nodes_set):
+                assert i == item
+        else:
```

Comment:
in this case, homograph is created so `ItemSet` is always expected. please remove `if-else` and always assert on what to be expected directly.

Reply:
It is suggested by @frozenbugs that it is aceeptable for a homograph to generate a `ItemSetDict` with a single item in it.